app/src/main/java/com/pajato/android/gamechat/exp/fragment/ExpEnvelopeFragment.java

```diff
@@ -71,6 +73,18 @@
         Object payload = event.view.getTag();
         if (payload == null || !(payload instanceof MenuEntry)) return;
 
+        // Handle invitation - extend app invitation, dismiss menu and return (there is no
+        // new experience to start).
+        if (((MenuEntry) payload).titleResId == R.string.InviteFriendFromExpEnv) {
+            InvitationManager.instance.extendAppInvitation(getActivity(), mExperience.getGroupKey());
+            FabManager.game.dismissMenu(this);
+            return;
+        } else if (((MenuEntry) payload).titleResId == R.string.InviteFriendFromChat ||
```

Comment:
I take it back, now I see what is going on.  it still might be written in a simpler way.  I'd be surprised in AS doesn't offer up some guidance.